### PR TITLE
Add primitive registry for ItemColorMappers

### PR DESCRIPTION
**Pull request by SquidDev**
This simply builds up a mapping of `Item` -> `ItemColorMapper`, and then registers them into the backing map when it becomes available.

### CODE REVIEW

**Comment by asiekierka** [comment]
I was considering this!

However, a part of me was considering the issue of using existing ItemColorMappers and BlockColorMappers in ways. In fact, the same goes for the current RendererRegistries - the current way doesn't let you really "copy" renderers/mappers.

**Comment by SquidDev** [comment]
So I've tried to interoperate the things discussed in IRC and Discord, though I'm honestly not sure how much nicer this is. Please comment everyone, because the more I look it, the more uncertain I am of anything!

File: src/main/java/net/fabricmc/fabric/mixin/client/render/MixinItemColorMap.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(ItemColorMap.class)
+public class MixinItemColorMap implements ColorMapperRegistryImpl.ColorMapperHolder<ItemContainer, ItemColorMapper> {
+	@Shadow
+	@Final
+	private IdList<ItemColorMapper> field_1996;
```
Comment (darkhax) [inline src/main/java/net/fabricmc/fabric/mixin/client/render/MixinItemColorMap.java]: I feel we should only be patching things that have been mapped. It makes the patches easier to read, and from what I understand once Yarn has a name in place this patch would need to be updated.

Comment (SquidDev) [reply src/main/java/net/fabricmc/fabric/mixin/client/render/MixinItemColorMap.java]: Yeah, that's not a bad shout. I'll put together a PR now - I don't think there's an existing one.

Comment (asiekierka) [reply src/main/java/net/fabricmc/fabric/mixin/client/render/MixinItemColorMap.java]: >from what I understand once Yarn has a name in place this patch would need to be updated.

@darkhax Not anymore! This only ever affected refmap entries which were method targets, I fixed this very recently!

File: src/main/java/net/fabricmc/fabric/api/client/render/ColorMapperRegistry.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+package net.fabricmc.fabric.api.client.render;
+
+import net.fabricmc.fabric.impl.client.render.ColorMapperRegistryImpl;
+import net.minecraft.block.Block;
+import net.minecraft.client.render.block.BlockColorMapper;
+import net.minecraft.client.render.item.ItemColorMapper;
+import net.minecraft.item.ItemContainer;
+
+public interface ColorMapperRegistry<T, Mapper> {
+	ColorMapperRegistry<ItemContainer, ItemColorMapper> ITEMS = ColorMapperRegistryImpl.ITEMS;
```
Comment (asiekierka) [inline src/main/java/net/fabricmc/fabric/api/client/render/ColorMapperRegistry.java]: Hmm. One minor nitpick - we do Registry.ITEM, so why not ColorMapperRegistry.ITEM?

File: src/test/java/net/fabricmc/fabric/Run.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (asiekierka) [inline src/test/java/net/fabricmc/fabric/Run.java]: You surely didn't want to push this?

Comment (SquidDev) [reply src/test/java/net/fabricmc/fabric/Run.java]: Curses. I could have sworn I had gitignored this file.

**Comment by SquidDev** [comment]
Fantastic, thank you all!

**Pull request merged**